Upgrade contribution policy and AI usage (#4989)
lima-vm/lima: .github/PULL_REQUEST_TEMPLATE.md

Reviewer: Could this be placed in https://github.com/lima-vm/.github

Author: Done in https://github.com/lima-vm/.github/pull/9

<!-- Please read our contribution rules before submitting:

- Pull requests guide: https://lima-vm.io/docs/community/contributing/#sending-pull-requests
- AI usage policy: https://lima-vm.io/docs/community/contributing/#ai-contribution-rules

-->

## What This PR Changes

<!-- Explain the single problem this PR fixes, in your own words. -->

## Linked Issue (Required in most cases)

> <!-- Link the issue here. The issue must be approved by a maintainer ("yes" or ready-to-work label), unless your PR is an allowed exception. -->

Reviewer: ```suggestion
<!-- Link the issue here. If your issue is not approved by a maintainer, don't expect your PR to be merged. -->
```

It is ok to submit a PR ahead of approval.
They just have to keep in mind that they may end up in wasting their time, LLM token cost, etc.
